main.py

```diff
@@ -83,34 +91,93 @@ def run(driver, shoe_type, username, password, url, shoe_size, login_time=None,
                 LOGGER.info("Page load timed out but continuing anyway")
             
             try:
-                select_shoe_size(driver=driver, shoe_size=shoe_size, shoe_type=shoe_type)
+                skip_size_selection = False
+                substring = "size="
+                if substring in url:
+                    skip_size_selection = True
+                select_shoe_size(driver=driver, shoe_size=shoe_size, shoe_type=shoe_type, skip_size_selection=skip_size_selection)
             except Exception as e:
-                # Try refreshing page since you can't click Buy button without selecting size
+                # Try refreshing page since you can't click Buy button without selecting size (except if size parameter passed in)
                 LOGGER.exception("Failed to select shoe size: " + str(e))
                 continue
-
+                
             try:
                 click_buy_button(driver=driver)
             except Exception as e:
-                LOGGER.exception("Failed to click buy button: " + str(e))
+                LOGGER.exception("Failed to click buy button: " + str(e))                                
                 six.reraise(Exception, e, sys.exc_info()[2])
-
-            if select_payment:
+                
+            skip_add_address, skip_select_shipping, skip_payment = poll_checkout_phase_one(driver=driver, skip_add_address=skip_add_address, skip_select_shipping=skip_select_shipping, skip_payment=skip_payment)
```

Comment:
`skip_add_address`, `skip_select_shipping`, and `skip_payment` should not be params of `poll_checkout_phase_one`. They are always False when passed in so they can just be local variables inside that function.